src/logic/player.cc

```diff
@@ -1479,14 +1484,52 @@ void Player::update_vision(const Area<FCoords>& area, bool force_visible) {
 	if (egbase().objects().is_cleaning_up()) {
 		return;
 	}
+
+	// Build a list of nearby objects.
+	// This way we only evaluate all objects (which is very expensive) once.
+	// When updating individual fields, only the objects from this list are evaluated.
+	std::list<std::pair<int, const MapObject*>> nearby_objects;
```

Comment:
Pull the code to build the list out into a new function `seers_for(FCoords)`

Reply:
Good idea. Great idea even because it helped me find a bug :-)